lasagne/init.py

```diff
@@ -50,50 +50,77 @@ def sample(self, shape):
 
 
 class Glorot(Initializer):
-    def __init__(self, initializer=Normal, gain=1.0):
+    def __init__(self, initializer=Normal, gain=1.0, c01b=False):
         if gain == 'relu':
             gain = np.sqrt(2)
 
         self.initializer = initializer
         self.gain = gain
+        self.c01b = c01b
 
     def sample(self, shape):
-        n1, n2 = shape[:2]
-        receptive_field_size = np.prod(shape[2:])
+        if self.c01b:
+            if len(shape) != 4:
+                raise RuntimeError(
```

Comment:
Hmm, I though that would be a ValueError. `shape` is of correct type, but inappropriate value. Or `c01b` is of correct type, but inappropriate value. But actually it's just the combination thats inappropriate, so the [descriptions](https://docs.python.org/2/library/exceptions.html) don't fit exactly. What would numpy do, or Theano?